staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -660,19 +736,24 @@ func (p *sharedProcessor) run(stopCh <-chan struct{}) {
 	func() {
 		p.listenersLock.RLock()
 		defer p.listenersLock.RUnlock()
-		for _, listener := range p.listeners {
+		for listener := range p.listeners {
 			p.wg.Start(listener.run)
 			p.wg.Start(listener.pop)
 		}
 		p.listenersStarted = true
 	}()
 	<-stopCh
-	p.listenersLock.RLock()
-	defer p.listenersLock.RUnlock()
-	for _, listener := range p.listeners {
+
+	p.listenersLock.Lock()
```

Comment:
How long does that Wait() take on line 752? Does this lock get held the entire time while the informer is in operation?

Reply:
I've removed this as a possible issue by placing the locking in an anonymous function scope, so the wait happens without holding any lock